### PR TITLE
Add web-dev container and update to React 18

### LINKED ISSUE #8: Use docker compose to run locally
Instead of only running backend in containers, a [docker-compose](https://docs.docker.com/compose/) should be more accessible and easier to the end-user, since they will need docker to run the backend anyway. That should even help when developing frontend.

What needs to be done in order to achieve it:  
* Create a dockerfile that builds the project and then runs it.
* Create a docker-compose file that runs both the backend and then the frontend, all in the same network. In the future, the backend application can be in more than one network, to be in a separate network with only backend app and frontend. 

With this, you should be able to run all applications (in the proper order) with `docker-compose up`, creating a docker container for each service instead of doing that with separate commands and the application in the shell session.

### PR DESCRIPTION
**Pull request by Henriquelay**
Resolves #8.

Big size update. :truck: 

This creates a web-development container, with update React 18 and all dependencies, updated and balated `.eslintrc.yaml`.
Lots of things had to change to update to React 18. Mostly, function components now don't return `React.FC`, and instead it all uses a generic `JSX.Element`. Also, no need to import from `React` everytime :tada:.

I had to do both actions at the same time because of some errors I was getting with installing React on the container, and updating solved it (and caused a bunch of new others, obviously).

Also added configs for Thunder (HTTP) Client.

I have continued on the same branch as before because the issue has been reopened, but thinking about it now, maybe it wasn't a good idea. But it's too late.

### CODE REVIEW

**Comment by Henriquelay** [comment]
Shit, I marked to merge to master.

**Comment by Henriquelay** [comment]
There is no conflict, GitHub is drunk. 
image

image

File: web/src/routes/index.tsx [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
 
-    <Route path="/login" component={SignIn} />
-    <Route path="/registrar" component={SignUp} />
-    <Route path="/conta" component={Account} isPrivate />
-    <Route path="/notificacoes" component={Notifications} />
-    <Route path="/privacidade" component={ToS} />
-  </Switch>
-)
+  // If autenticated context is present
+  return user ? <Outlet /> : <Navigate to="/login" />;
```
Comment (AtilioA) [inline web/src/routes/index.tsx]: The `/notificacoes` route should also be private. The rest of the pages should only have visual/functional differences, except for Sign(In|Up) which should redirect the user to the dashboard when they're authenticated.

**Comment by AtilioA** [comment]
I don't know if this private stuff is working (couldn't reproduce) but LGTM :+1:

**Comment by Henriquelay** [comment]
I'll resolve the merge conflicts later today then we can finally merge it!

**Comment by Henriquelay** [comment]
I have merged manually through the GitHub web editor and have not tested yet, so there is _exactly_ zero chance this goes wrong.

**Comment by AtilioA** [comment]
Seems good to me now, but we're missing Toast messages and feedback is totally bonkers right now

**Comment by Henriquelay** [comment]
> we're missing Toast messages

Toast messages were never there, were they?

**Comment by AtilioA** [comment]
bruh i can't scroll

**Comment by Henriquelay** [comment]
I can't take it anymore 🤠

**Pull request merged**